apollo-router/src/services/layers/persisted_queries/manifest_poller.rs

```diff
@@ -490,46 +508,69 @@ async fn add_chunk_to_operations(
     operations: &mut PersistedQueryManifest,
     http_client: Client,
 ) -> Result<(), BoxError> {
-    // TODO: chunk URLs will eventually respond with fallback URLs, when it does, implement falling back here
-    if let Some(chunk_url) = chunk.urls.get(0) {
-        let chunk = http_client
-            .get(chunk_url.clone())
-            .send()
-            .await
-            .and_then(|r| r.error_for_status())
-            .map_err(|e| -> BoxError {
-                format!(
-                    "error fetching persisted queries manifest chunk from {}: {}",
-                    chunk_url, e
-                )
-                .into()
-            })?
-            .json::<SignedUrlChunk>()
-            .await
-            .map_err(|e| -> BoxError {
-                format!(
-                    "error reading body of persisted queries manifest chunk from {}: {}",
-                    chunk_url, e
-                )
-                .into()
-            })?;
-
-        if chunk.format != "apollo-persisted-query-manifest" {
-            return Err("chunk format is not 'apollo-persisted-query-manifest'".into());
-        }
-
-        if chunk.version != 1 {
-            return Err("persisted query manifest chunk version is not 1".into());
+    let mut it = chunk.urls.iter().peekable();
+    while let Some(chunk_url) = it.next() {
+        match fetch_chunk(http_client.clone(), chunk_url).await {
+            Ok(chunk) => {
+                for operation in chunk.operations {
+                    operations.insert(operation.id, operation.body);
+                }
+                return Ok(());
+            }
+            Err(e) => {
+                if it.peek().is_some() {
```

Comment:
This is a different pattern from how `stream_from_uplink` (or specifically, `fetch`) handles errors.

In `stream_from_uplink`, individual failures are logged to `debug` with a message specifying "Other endpoints will be tried" (even if it's the last one!), and if they all fail, a constant `Err` without the specific error message in it is returned. My understanding is that we don't want to spam stderr with single-endpoint errors, but this does mean that the eventual error doesn't have specific details on the problem that occurred (unless you go and reproduce with the log level set to debug).

Here, we do the `debug`-with-"Other endpoints will be tried" thing only when there actually are more URLs to try; errors from the last URL are returned directly.

I think this is better than the existing approach because it's nice to actually include information about some failure in the eventual error, but I'm happy to rewrite this to work more like `fetch` if you'd like.